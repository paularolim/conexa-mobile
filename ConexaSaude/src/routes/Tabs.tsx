import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { useCallback } from 'react';
import { useTheme } from 'styled-components/native';

import { Icon } from '@components/Icon';
import { CreateAppointment } from '@screens/CreateAppointment';
import { Home } from '@screens/Home';
import { Profile } from '@screens/Profile';

import { CustomTabBarButton } from './components/CustomTabBarButton';
import { styles } from './styles';
import { TabBarIconProps, TabsRoutesParamList } from './types';

const Tab = createBottomTabNavigator<TabsRoutesParamList>();

export function Tabs() {
  const { colors } = useTheme();

  const colorRule = useCallback(
    (focused: boolean) => (focused ? colors.pink : colors.gray),
    [colors.gray, colors.pink],
  );

  const HomeIcon = useCallback<TabBarIconProps>(
    ({ focused }) => <Icon name="house" width={32} height={32} fill={colorRule(focused)} />,
    [colorRule],
  );

  const PlusIcon = useCallback<TabBarIconProps>(
    () => <Icon name="plus" width={32} height={32} fill={colors.white} />,
    [colors.white],
  );

  const UserIcon = useCallback<TabBarIconProps>(
    ({ focused }) => <Icon name="user" width={32} height={32} fill={colorRule(focused)} />,
    [colorRule],
  );

  return (
    <Tab.Navigator
      screenOptions={{
        headerShown: false,
        tabBarShowLabel: false,
        tabBarStyle: { ...styles.tabBarShadow, ...styles.tabBarStyle },
      }}
    >
      <Tab.Screen name="Home" component={Home} options={{ tabBarIcon: HomeIcon }} />
      <Tab.Screen
        name="CreateAppointment"
        component={CreateAppointment}
        options={{
          tabBarIcon: PlusIcon,
          tabBarButton: CustomTabBarButton,
        }}
      />
      <Tab.Screen name="Profile" component={Profile} options={{ tabBarIcon: UserIcon }} />
    </Tab.Navigator>
  );
}
