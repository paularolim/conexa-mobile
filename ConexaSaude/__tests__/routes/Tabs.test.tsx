import { NavigationContainer } from '@react-navigation/native';
import { cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { Tabs } from '@routes/Tabs';

import { Wrapper } from '../Wrapper';

jest.useFakeTimers();

describe('Tabs', () => {
  afterEach(() => cleanup());

  it('should render routes correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <NavigationContainer>
            <Tabs />
          </NavigationContainer>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
