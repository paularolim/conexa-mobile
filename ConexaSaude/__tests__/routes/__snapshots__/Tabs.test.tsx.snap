// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render routes correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenNavigationContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onGestureCancel={[Function]}
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            [
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              Home
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#2ED47A",
                  "borderRadius": 4,
                  "justifyContent": "center",
                  "maxWidth": "100%",
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                    {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                    },
                  ]
                }
                type="filled"
              >
                Consulta
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "white",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0,
        "bottom": 0,
        "elevation": 5,
        "height": 60,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "shadowColor": "#898989",
        "shadowOffset": {
          "height": -5,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.5,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": true,
          }
        }
        accessibilityStates={
          [
            "selected",
          ]
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgHouse"
              height={32}
              stroke="#F28080"
              width={32}
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgHouse"
              height={32}
              stroke="#707683"
              width={32}
            />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="CreateAppointment, tab, 2 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#F28080",
            "borderRadius": 12,
            "elevation": 50,
            "height": 50,
            "opacity": 1,
            "shadowColor": "#f28080",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 5,
            "top": 5,
            "width": 70,
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgPlus"
              fill="#FFFFFF"
              height={32}
              width={32}
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgPlus"
              fill="#FFFFFF"
              height={32}
              width={32}
            />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Profile, tab, 3 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgUser"
              height={32}
              stroke="#F28080"
              width={32}
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgUser"
              height={32}
              stroke="#707683"
              width={32}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
